packages/nodes-base/nodes/Google/Sheet/v2/actions/router.ts

```diff
@@ -49,16 +50,17 @@ export async function router(this: IExecuteFunctions): Promise<INodeExecutionDat
 					sheetName = await googleSheet.spreadsheetGetSheetNameById(sheetId);
 			}
 
-			operationResult.push(
-				...(await sheet[googleSheets.operation].execute.call(
-					this,
-					googleSheet,
-					sheetName,
-					sheetId,
-				)),
+			results = await sheet[googleSheets.operation].execute.call(
+				this,
+				googleSheet,
+				sheetName,
+				sheetId,
 			);
 		} else if (googleSheets.resource === 'spreadsheet') {
-			operationResult.push(...(await spreadsheet[googleSheets.operation].execute.call(this)));
+			results = await spreadsheet[googleSheets.operation].execute.call(this);
+		}
+		if (results?.length) {
+			operationResult = operationResult.concat(results);
```

Comment:
I ran a small benchmark comparing `.concat`, pushing items from `results` into `operationResult` one at a time, pushing in batches of 1k, and pushing in batches of 100k.
`.concat` was consistently many times faster, while also produced the least amount of garbage for the GC to worry about.